cli/command_maintenance_set.go

```diff
@@ -121,6 +124,22 @@ func (c *commandMaintenanceSet) setMaintenanceEnabledAndIntervalFromFlags(ctx co
 	}
 }
 
+func (c *commandMaintenanceSet) setMaintenanceObjectLockExtendFromFlags(ctx context.Context, p *maintenance.Params, changed *bool) {
+	// we use lists to distinguish between flag not set
+	// Zero elements == not set, more than zero - flag set, in which case we pick the last value
+	if len(c.extendObjectLocks) > 0 {
+		lastVal := c.extendObjectLocks[len(c.extendObjectLocks)-1]
+		p.ExtendObjectLocks = lastVal
+		*changed = true
+
+		if lastVal {
+			log(ctx).Info("Object Lock extension maintenance enabled.")
+		} else {
+			log(ctx).Info("Object Lock extension maintenance disabled.")
+		}
+	}
+}
```

Comment:
There's a problematic situation here: If the full maintenance window interval (`FullCycle.Interval`) is longer than, equal to, or even slightly less than (it takes time to issue the extensions), the retention period (`blobcfg.RetentionPeriod`), the blobs will NOT be protected, and the user might not be aware of this.

One possible solution could be:
- Calculate an acceptable relationship between full maint interval and retention period (a trivial example would be retention period must be, say, twice as long as the maintenance interval).
- Refuse to set `ExtendObjectLocks` here the current interval and period settings do not satisfy the above relationship. Throw a noisy error instead.
- Similarly, if `ExtendObjectLocks` is enabled, refuse any changes to the maintenance interval that would result in a violation of the above relationship. Noisy error.
- And same for retention period changes - if `ExtendObjectLocks` is enabled, throw a noisy error if the change would violate it.

Basically prevent the system from getting into that unsafe state altogether by policing changes to these three settings. Best if we can prevent users from unwittingly shooting themselves in the foot.